src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -1236,10 +1238,32 @@ void CodegenNeuronCppVisitor::print_nrn_jacob() {
     printer->add_newline(2);
     printer->add_line("/** nrn_jacob function */");
 
-    printer->fmt_line(
+    printer->fmt_push_block(
         "static void {}(_nrn_model_sorted_token const& _sorted_token, NrnThread* "
-        "_nt, Memb_list* _ml_arg, int _type) {{}}",
-        method_name(naming::NRN_JACOB_METHOD));
+        "_nt, Memb_list* _ml_arg, int _type)",
+        method_name(naming::NRN_JACOB_METHOD));  // begin function
+
+    printer->add_multi_line(
+        "_nrn_mechanism_cache_range _lmr{_sorted_token, *_nt, *_ml_arg, _type};");
+
+    printer->fmt_line("auto inst = make_instance_{}(_lmr);", info.mod_suffix);
+    printer->fmt_line("auto node_data = make_node_data_{}(*_nt, *_ml_arg);", info.mod_suffix);
+    printer->fmt_line("auto nodecount = _ml_arg->nodecount;");
+    printer->push_block("for (int id = 0; id < nodecount; id++)");  // begin for
+
+    for (const auto& variable: codegen_float_variables) {
+        if (variable->get_name() ==
+                std::string(nmodl::codegen::naming::CONDUCTANCE_UNUSED_VARIABLE) ||
+            variable->get_name() == std::string(nmodl::codegen::naming::CONDUCTANCE_VARIABLE)) {
```

Comment:
Note to reviewer: should we only use `CONDUCTANCE_UNUSED_VARIABLE`, or also `CONDUCTANCE_VARIABLE` (when is each one generated/used)?

Reply:
If we follow CoreNEURON (for now), we don't need the conductance at all. If you compare, it doesn't store `g` computed in one of the currents anywhere. For debugging purposes it can store the value in `g_unused`.

NEURON  is different, it stores it to retrieve it in the function `*_jacob_*`. If you want to also implement that part we would probably want to understand why there's this difference. One explanation could be that it's needed for variable timestepping. If we need to store it then the correct place to store would be in a range variable called `_g` (same as NEURON uses), probably by renaming `g_unused` to `_g` when compiling for NRN.

Reply:
I think by looping over the `codegen_float_variables` you risk that a mod file defines a variable that happens to be called `"g"` (aka `nmodl::codegen::naming::CONDUCTANCE_VARIABLE`), but with the meaning that `"g"` is whatever they define it to be in that MOD file; and then you overwrite with something the MOD file didn't expect.